signer/src/storage/postgres.rs

```diff
@@ -811,6 +1002,67 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn get_deposit_request_report(
```

Comment:
Yeah, I like it. I figure that if it's useful then we'll move it over. Since I've been skipping the in-memory implementations here anyway, I suppose there is no cost to just moving it over.

I'm also refactoring this around in an upcoming PR. So maybe wait and see if you like the changes from there first?

Reply:
Sounds like we're on the same page anyway so I'm happy :)